api/http/handler/docker/dashboard.go

```diff
@@ -0,0 +1,166 @@
+// @summary Get counters for the dashboard
+// @description **Access policy**: restricted
+// @tags docker
+// @security jwt
+// @param environmentId path int true "Environment identifier"
+// @accept json
+// @produce json
+// @success 200 {object} dashboardResponse "Success"
+// @failure 400 "Bad request"
+// @failure 500 "Internal server error"
+// @router /docker/{environmentId}/dashboard [post]
+func (h *Handler) dashboard(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	cli, httpErr := utils.GetClient(r, h.dockerClientFactory)
+	if httpErr != nil {
+		return httpErr
+	}
+
+	containers, err := cli.ContainerList(r.Context(), container.ListOptions{All: true})
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve Docker containers", err)
+	}
+
+	containerStats := docker.CalculateContainerStats(containers)
+
+	images, err := cli.ImageList(r.Context(), image.ListOptions{})
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve Docker images", err)
+	}
+
+	var totalSize int64
+	for _, image := range images {
+		totalSize += image.Size
+	}
+
+	info, err := cli.Info(r.Context())
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve Docker info", err)
+	}
+
+	isSwarmManager := info.Swarm.ControlAvailable && info.Swarm.NodeID != ""
+	serviceCount := 0
+	var services []swarm.Service
+	if isSwarmManager {
+		servicesRes, err := cli.ServiceList(r.Context(), types.ServiceListOptions{})
+		if err != nil {
+			return httperror.InternalServerError("Unable to retrieve Docker services", err)
+		}
+
+		serviceCount = len(services)
+		services = servicesRes
+	}
+
+	volumes, err := cli.VolumeList(r.Context(), volume.ListOptions{})
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve Docker volumes", err)
+	}
+
+	networks, err := cli.NetworkList(r.Context(), types.NetworkListOptions{})
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve Docker networks", err)
+	}
+
+	environment, err := middlewares.FetchEndpoint(r)
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve environment", err)
+	}
+
+	user, err := security.RetrieveUserFromRequest(r, h.dataStore)
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve user", err)
+	}
+
+	stacksCount, err := countStacks(environment, user, h, containers, isSwarmManager, services)
+	if err != nil {
+		return httperror.InternalServerError("Unable to retrieve stacks", err)
+	}
+
+	return response.JSON(w, dashboardResponse{
+		Containers: containerStats,
+		Images: imagesCounters{
+			Total: len(images),
+			Size:  totalSize,
+		},
+		Services: serviceCount,
+		Volumes:  len(volumes.Volumes),
+		Networks: len(networks),
+		Stacks:   stacksCount,
+	})
+}
+
+func countStacks(environment *portainer.Endpoint, user *portainer.User, h *Handler, containers []types.Container, isSwarmManager bool, services []swarm.Service) (int, error) {
```

Comment:
I think this is too complex.
first, pretty sure that we don't need to count services, as services spawn containers, so docker swarm stacks will appear also as containers (just need to count for duplicates). see swarm snapshot stacks page.

we also need to check that external stacks are not in the list of endpoint stacks